corehq/apps/app_manager/models.py

```diff
@@ -4521,6 +4523,10 @@ def make_build(self, comment=None, user_id=None):
             # but check explicitly so as not to change the _id if it exists
             copy._id = uuid.uuid4().hex
 
+        errors = copy.validate_app()
+        if errors:
+            raise AppValidationError(errors)
```

Comment:
Is there any kind of cleanup to do here? What prevents `copy` from showing up on the releases page?

Reply:
`copy` is actually not saved until later, outside `make_build`.